File: src/dataset/Java/anagram_Exercism_java_rabestro/gemini/GeminiFlash_Anagram_v2.java
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

class Anagram {
    private final String sourceWordLower;
    private final char[] sortedChars;

    Anagram(final String word) {
        sourceWordLower = word.toLowerCase();
        sortedChars = sourceWordLower.toCharArray();
        Arrays.sort(sortedChars);
    }

    List<String> match(final List<String> possibleAnagrams) {
        return possibleAnagrams.stream()
                .filter(this::isAnagram)
                .collect(Collectors.toUnmodifiableList());
    }

    private boolean isAnagram(final String otherWord) {
        if (sourceWordLower.length() != otherWord.length()) {
            return false;
        }

        String otherWordLower = otherWord.toLowerCase();

        if (sourceWordLower.equals(otherWordLower)) {
            return false;
        }

        char[] otherSortedChars = otherWordLower.toCharArray();
        Arrays.sort(otherSortedChars);

        return Arrays.equals(sortedChars, otherSortedChars);
    }
}

File: src/dataset/typescript/anagram_exercism-javascript-samajammin/gemini/GeminiFlash_anagram_v2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class Anagram {
    constructor(anagram) {
        this.anagram = anagram.toLowerCase();
        this.sortedAnagram = [...this.anagram].sort().join('');
    }
    isAnagram(word) {
        if (this.anagram === word) {
            return false;
        }
        if (this.anagram.length !== word.length) {
            return false;
        }
        return this.sortedAnagram === [...word].sort().join('');
    }
    matches(...words) {
        return words.filter((word) => this.isAnagram(word.toLowerCase()));
    }
}
exports.default = Anagram;
